### PR TITLE
ceph-dencoder: add missing begin_iter & end_iter item for RGWObjManifest

### PR DESCRIPTION
**Pull request by cooboos**
Signed-off-by: Songbo Wang wangsongbo@cloudin.cn

### CODE REVIEW

File: src/rgw/rgw_json_enc.cc [hunk 1]
```diff
@@ -107,6 +127,9 @@ void RGWObjManifest::dump(Formatter *f) const
   ::encode_json("rules", rules, f);
   ::encode_json("tail_instance", tail_instance, f);
   ::encode_json("tail_placement", tail_placement, f);
+
+  f->dump_object("begin_iter", begin_iter);
```
Comment (yaozongyou) [inline src/rgw/rgw_json_enc.cc]: should be able to use encode_json() instead, probably better using it instead of all the direct formatter->dump() calls

Comment (cooboos) [reply src/rgw/rgw_json_enc.cc]: In my previous pr(#18959 ), @tchaikov suggest to use formatter's method.  It confused me which should be used. 
image

Comment (tchaikov) [reply src/rgw/rgw_json_enc.cc]: @yaozongyou well, it depends on if we are dumping a JSON or not. but in this case, i think `RGWObjManifest::dump()` is a method accepting a general `Formatter`, why shall we use `encode_json()`?

Comment (yaozongyou) [reply src/rgw/rgw_json_enc.cc]: @tchaikov encode_json's last parameter is a general ceph::Formatter，it has the same capability with formatter->dump. and in this source code file, as the original GWObjManifest::dump using encode_json, we'd better keeping consistent with it.

Comment (tchaikov) [reply src/rgw/rgw_json_enc.cc]: > it has the same capability with formatter->dump.

also please note that the same applies to to `encode_json()`. and the latter is a wrapper of the `formatter->dump()`, extra cpu cycles wasted if not inlined.

> as the original GWObjManifest::dump using encode_json, we'd better keeping consistent with it.

`Foo::dump()` is a general interface used by `Formatter` not just JSON dump methods, it's consistent to follow this convention. but i understand, the to use `encode_json()` is also consistent with the rest part of this method or this source file. so no strong feelings either way. the reason i am chiming in here is merely that i am @'ed.

Comment (yaozongyou) [reply src/rgw/rgw_json_enc.cc]: as @tchaikov has no favor for either way and @mattbenjamin has approved it, @cooboos let's keep it as it is.

Comment (cooboos) [reply src/rgw/rgw_json_enc.cc]: @yaozongyou ok， thank you !

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-01-08_13:52:40-rados-wip-kefu-testing-2018-01-08-1914-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-01-09_13:55:16-rados-wip-kefu-testing-2018-01-09-1600-distro-basic-mira/

**Comment by tchaikov** [comment]
retest this please

**Comment by tchaikov** [comment]
retest this please.

**Comment by mattbenjamin** [comment]
@cooboos are you adding the iterator format for debugging or tracing purposes? I think that's it's only utility, but adding @yehudasa to cross check.

**Comment by yuriw** [comment]
test this please

**Comment by yuriw** [comment]
[wip-yuri3-testing](https://trello.com/c/jMbaJHhX/299-wip-yuri3-testing)

**Pull request merged**